### ISSUE TITLE
Division by 0 does not raise ZeroDivisionError

### ISSUE THREAD

**Issue by cooldannyboy** [cython/cython#2820]
I compile a python file test.py to a shared library by Cython, then use python to import this shared library for testing.The original python file raise ZeroDivisionError but this shared library doesn't.The following is the whole code to reproduce this issue

test.py:
```
try:
    print("try")
    l = [0.0]
    ll = [1.0/x for x in l]
    print("try end")

except ZeroDivisionError:
    print("exception")
```

Original python output:
try
exception

Shared library by Cython output:
try
try end

python version: 3.6
cython version: 0.29.3

**Comment by scoder** [comment]
Thanks for the report. You can't catch it because it does not get raised. :)
It's an incomplete optimisation. I'll look into it.

**Comment by scoder** [comment]
Fixed in Cython 0.29.4.
AFAICT, there is no work-around for older versions.
